org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceExecuteCommandHandler.java

```diff
@@ -219,4 +221,29 @@ public void handleException(Throwable ex) {
 		});
 		return resultValues[0];
 	}
+
+	@Override
+	public void added(IExtension[] extensions) {
+		Set<DelegateCommandHandlerDescriptor> addedDescriptors = Stream.of(extensions)
+				.filter(extension -> extension.getExtensionPointUniqueIdentifier().equals(EXTENSION_POINT_ID))
+				.flatMap(extension -> Stream.of(extension.getConfigurationElements()))
+				.map(configurationElement -> new DelegateCommandHandlerDescriptor(configurationElement))
+				.collect(Collectors.toSet());
+		fgContributedCommandHandlers.addAll(addedDescriptors);

```

Comment:
As requested, but I changed the cache to use a Map with a "id of the descriptor", since I cann't predict how expensive is creating a `DelegateCommandHandlerDescriptor` just for removing it from the cache.